clippy_lints/src/utils/conf.rs

```diff
@@ -387,6 +387,11 @@ define_Conf! {
     ///
     /// The maximum size of the `Err`-variant in a `Result` returned from a function
     (large_error_threshold: u64 = 128),
+    /// Lint::MUTABLE_KEY
+    ///
+    /// A list of paths to types that should be treated like `Arc`, i.e. ignored but
+    /// for the generic parameters for determining interior mutability
+    (arc_like_types: Vec<String> = Vec::new()),
```

Comment:
Seems like a reasonable escape hatch, I think the name might be better as a bit more direct, `ignore_interior_mutability` or something along those lines as there may be other types that aren't `Arc`-like

What do you think about setting the default to `vec!["bytes::Bytes"]` to cover the seemingly most popular false positive?